client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/index.tsx

```diff
@@ -22,23 +22,55 @@ import type { SiteSelect } from '@automattic/data-stores';
 
 import './style.scss';
 
-type LaunchpadProps = {
+// Offloads whether to use siteSlug or siteId to useSite() enabling the launchpad to use either query param
+const LaunchpadSiteWrapper: Step = ( {
+	navigation,
+	flow,
+}: {
 	navigation: NavigationControls;
 	flow: string | null;
+} ) => {
+	const site = useSite();
+
+	if ( ! site || ! flow ) {
+		return null;
+	}
+
+	return (
+		<Launchpad
+			site={ site }
+			siteId={ site.ID }
+			siteSlug={ urlToSlug( site.URL ) }
+			navigation={ navigation }
+			flow={ flow }
+		/>
+	);
 };
 
-const Launchpad: Step = ( { navigation, flow }: LaunchpadProps ) => {
+function Launchpad( {
+	siteSlug,
```

Comment:
I think we don't need the slug anymore. Everything that uses the slug most probably works with site ID.

Reply:
They do but I'd like to keep using it for the site specific /home etc routes.